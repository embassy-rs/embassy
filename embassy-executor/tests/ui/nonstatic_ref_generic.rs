#![cfg_attr(feature = "nightly", feature(impl_trait_in_assoc_type))]

#[embassy_executor::task]
async fn foo<'a>(_x: &'a u32) {}

fn main() {}
