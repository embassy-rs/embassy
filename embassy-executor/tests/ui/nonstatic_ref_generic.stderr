error: task functions must not be generic
 --> tests/ui/nonstatic_ref_generic.rs:4:1
  |
4 | async fn foo<'a>(_x: &'a u32) {}
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Arguments for tasks must live forever. Try using the `'static` lifetime.
 --> tests/ui/nonstatic_ref_generic.rs:4:23
  |
4 | async fn foo<'a>(_x: &'a u32) {}
  |                       ^^
